### ISSUE TITLE
Emacs doesn't use Spack's terminfo files

### ISSUE THREAD

**Issue by hartzell** [spack/spack#8353]
**TL;DR** -- If you have been building `emacs +tls`, the changes in #8302 *might* have changed where emacs looks for terminfo files.  If you want to go back to the old behavior, you can build `emacs ^ncurses+termlib`.

---

I'm playing around with Emacs configurations (cough -- rabbit hole -- cough) and discovered that emacs isn't using terminfo entries from Spack's ncurses package (e.g. `tmux-256color`).

As I dug into it a bit I noticed that my Spack-supplied `git` was behaving poorly (as is `bash`), at which point I decided that I should either stick with terminfo names that are in the system database or tuck a special entry in `~/.terminfo`.

But, in case someone else trips over this, I thought I'd write it up for posterity.

If I build `emacs ^ncurses+termlib` it *does* read those entries, which is confusing.  None of the options that are passed to configure when `+termlib` seem to have anything to do with where the terminfo entries is searched for.

Here are the switches:

```
            opts.extend(('--with-termlib',
                         '--enable-termcap',
                         '--enable-getcap',
                         '--enable-tcap-names'))
```

and here are the associated help messages from `configure --help`:

```
            # generate separate terminfo library
            # compile in termcap fallback support
            # fast termcap load, no xrefs to terminfo
            # compile with user-definable terminal capabilities
```

Apparently what's happening is if Spack doesn't build a separate terminfo library then emacs ends up linking against the system's `libtinfo.so`. 

When I build emacs with an `ncurses +termlib` I don't see any signs that emacs is using anything from ncurses specifically, but it does use its `libtinfo`:

```
[hartzell@ip-10-96-1-107 ncurses-6.1]$ ldd $(which emacs)
	linux-vdso.so.1 =>  (0x00007fffe5c97000)
	librt.so.1 => /lib64/librt.so.1 (0x00007fa831040000)
	libxml2.so.2 => /home/hartzell/spack/opt/spack/linux-rhel7-x86_64/gcc-5.5.0/libxml2-2.9.8-hgfwbirg7su2pc67osk2fzrsilw22ybd/lib/libxml2.so.2 (0x00007fa830cd9000)
	libtinfo.so.6 => /home/hartzell/spack/opt/spack/linux-rhel7-x86_64/gcc-5.5.0/ncurses-6.1-rgdadtxyziks74sx7mzrpodj7ia6f36t/lib/libtinfo.so.6 (0x00007fa830a99000)
	libselinux.so.1 => /lib64/libselinux.so.1 (0x00007fa830872000)
	libgnutls.so.30 => /home/hartzell/spack/opt/spack/linux-rhel7-x86_64/gcc-5.5.0/gnutls-3.5.13-ju3fyaastivwai6ixccqwelwd7bhxuii/lib/libgnutls.so.30 (0x00007fa8304e6000)
	libpthread.so.0 => /lib64/libpthread.so.0 (0x00007fa8302ca000)
	libanl.so.1 => /lib64/libanl.so.1 (0x00007fa8300c6000)
	libm.so.6 => /lib64/libm.so.6 (0x00007fa82fdc4000)
	libz.so.1 => /home/hartzell/spack/opt/spack/linux-rhel7-x86_64/gcc-5.5.0/zlib-1.2.11-56rgaosb4ij2unhwfsnbiv7z3qjximce/lib/libz.so.1 (0x00007fa82fbac000)
	libc.so.6 => /lib64/libc.so.6 (0x00007fa82f7e9000)
	libdl.so.2 => /lib64/libdl.so.2 (0x00007fa82f5e4000)
	liblzma.so.5 => /home/hartzell/spack/opt/spack/linux-rhel7-x86_64/gcc-5.5.0/xz-5.2.4-xz3h7mccq4wkwdcl7twvmf2cpoawdwp7/lib/liblzma.so.5 (0x00007fa82f3be000)
	libpcre.so.1 => /home/hartzell/spack/opt/spack/linux-rhel7-x86_64/gcc-5.5.0/pcre-8.42-idvaazk4oqp6jj7sxfwq324jb7glms7m/lib/libpcre.so.1 (0x00007fa82f179000)
	/lib64/ld-linux-x86-64.so.2 (0x0000563d7ca1c000)
	libnettle.so.6 => /home/hartzell/spack/opt/spack/linux-rhel7-x86_64/gcc-5.5.0/nettle-3.3-gm37wrusxw523qx4pszjs3o56j5yqner/lib64/libnettle.so.6 (0x00007fa82ef43000)
	libhogweed.so.4 => /home/hartzell/spack/opt/spack/linux-rhel7-x86_64/gcc-5.5.0/nettle-3.3-gm37wrusxw523qx4pszjs3o56j5yqner/lib64/libhogweed.so.4 (0x00007fa82ed0e000)
	libgmp.so.10 => /home/hartzell/spack/opt/spack/linux-rhel7-x86_64/gcc-5.5.0/gmp-6.1.2-cd47nos5nivwx4zd5r37nxwy57e7qlb2/lib/libgmp.so.10 (0x00007fa82ea9a000)
	libintl.so.8 => /home/hartzell/spack/opt/spack/linux-rhel7-x86_64/gcc-5.5.0/gettext-0.19.8.1-rb545khehtpmessjbd56yjqzt6cycgra/lib/libintl.so.8 (0x00007fa82e88e000)
[hartzell@ip-10-96-1-107 ncurses-6.1]$
```

Perhaps the emacs'

```
  53     depends_on('ncurses')
```

is a bug and it should be `ncurses+termlib`?  Or perhaps it should not depend on it at all?

Deleting the line and building emacs (without `tls`) doesn't build `ncurses` and the resulting emacs seems to work.  Building `emacs +tls` builds ncurses by way of gnutls by way of gettext, but emacs doesn't seem to use it or care.
